### PR TITLE
completed RESTFUL APIs for the BusinessProfile model with testcases

### PR DESCRIPTION
**Pull request by Natiman58**

### CODE REVIEW

**Review by sgarciahelguera — COMMENTED** [review]
This looks good, but we should be using         "rest_framework_simplejwt.authentication.JWTAuthentication"

**Comment by sgarciahelguera** [comment]
I think we shouldn't add new auth methods (unless we decide to move to something else but if we are sticking to jwt we should test using it)

File: cubeseed/businessprofile/views.py [hunk 1]
```diff
@@ -7,5 +7,6 @@
 class BusinessProfileViewSet(viewsets.ModelViewSet):
     queryset = BusinessProfile.objects.all()
     serializer_class = BusinessProfileSerializer
-    permission_classes = [permissions.DjangoModelPermissions]
+    #permission_classes = [permissions.DjangoModelPermissions]
```
Comment (sgarciahelguera) [inline cubeseed/businessprofile/views.py:10]: I don't see why `permissions.IsAuthenticated` is better than  `permissions.DjangoModelPermissions` to run the tests (just curious) ? ?
I added an issue about permissions: #50 
We should probably merge this and worry about permissions later.

File: cubeseed/businessprofile/tests.py [hunk 2]
```diff
@@ -1,3 +1,104 @@
...
+        # Assert that the response status code is 200 OK
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Refresh the business profile from the test database and check if it's updated
+        self.business_profile.refresh_from_db()
+        self.assertEqual(self.business_profile.business_name, updated_data["business_name"])
+        self.assertEqual(self.business_profile.telephone, updated_data["telephone"])
+        self.assertEqual(self.business_profile.email, updated_data["email"])
+        self.assertEqual(self.business_profile.billing_address, updated_data["billing_address"])
+        self.assertEqual(self.business_profile.shipping_address, updated_data["shipping_address"])
```
Comment (sgarciahelguera) [inline cubeseed/businessprofile/tests.py:87]: There is an open pr #49 to replace addresses here and in user profiles, we can merge this as is and add an issue to replace those with a fk to the new entity

**Comment by sgarciahelguera** [comment]
I invited you to the backend team so that you can create a branch and pr from it next time.
The PR from a fork is the reason for the "create and publish docker ..." action to fail, it should work from a fork too (don't worry about it, I'll fix it at some point, we can ignore it for now as it will be run again on merge and shouldn't fail).

**Comment by Natiman58** [comment]
> Basic and Session:
> "rest_framework.authentication.BasicAuthentication",
> "rest_framework.authentication.SessionAuthentication",
> Are there just for swagger and django's router login, to help in development, but tests should not use them either

I used 'rest_framework.authentication.TokenAuthentication' to pass authenticated user token to the djago admin to get the API response the above methods were giving me 401 access denied error while testing ; we can only use it for testing purposes only and comment it out otherwise

File: cubeseed/businessprofile/views.py (hunk 1 above)
Comment (Natiman58) [reply cubeseed/businessprofile/views.py:10]: it's for passing updated data to django admin its a fix around the 403 error I got from permissions.DjangoModelPermissions while testing PUT route so we can comment it out after the testing

Comment (Natiman58) [reply cubeseed/businessprofile/views.py:10]: > it's for passing updated data to django admin its a fix around the 403 error I got from permissions.DjangoModelPermissions while testing PUT route so we can comment it out after the testing

> # Testing the PUT(update) businessprofile by id route
    def test_put_business_profile_by_id(self):
        # Get the URL for the specific business profile by its ID
        url = reverse("businessprofile-detail", args=[self.business_profile.id])

        # Data for updating the business profile
        updated_data = {
            "business_name": "Updated Business",
            "telephone": "9876543210",
            "email": "null",
            "billing_address": "00000",
            "shipping_address": "004400",
            "document_type": "PITIN"
        }

        # Make a PUT request to update the business profile
        response = self.client.put(url, updated_data, format="json", HTTP_AUTHORIZATION=self.auth_header)

        # Assert that the response status code is 200 OK
        self.assertEqual(response.status_code, status.HTTP_200_OK)

**Comment by sgarciahelguera** [comment]
You need to use a JWT token, checkout tests in userauth or address (this one is still in a pull request).
If we use a different authentication method for testing we cannot test the production application nor the jwt authentication.
Those are there for manual access to documentation and manual evaluation of methods, not for actual tests.

**Comment by sgarciahelguera** [comment]
Like this (copied from userauth tests):
```
response = self.client.post(reverse('token_obtain_pair'),{
             'username':'testuser',
            'password':'testpassword'
         })
         token = response.data['access']
         self.client.credentials(HTTP_AUTHORIZATION=f'Bearer {token}')
```

**Comment by sgarciahelguera** [comment]
Closing this, was replaced by #54

**Pull request closed without merging**